### PR TITLE
fix(#54): status improvements

### LINKED ISSUE #54: Make status messages more contextual
Currently, we have the same status message used for multiple cases which can be confusing at times.
For example when PR contains only "legit files", like `pom.xml` for which you most likely don't ship tests, it should have different msg than "There are some tests :)".

Additional improvements:

  * status context shouldn't contain quoted context (now it's `"ike-plugins"/"test-keeper" — There are some tests :)`)

### PR DESCRIPTION
**Pull request by bartoszmajsak**
- makes status context without extra quotes
- counts excluded files to treat it as different status message
- renames bot to alien-ike

Fixes #54

### CODE REVIEW

File: pkg/plugin/test-keeper/plugin/event_handler_gh_test.go [hunk 1]
```diff
@@ -235,7 +230,7 @@ var _ = Describe("Test Keeper Plugin features", func() {
 
 			gock.New("https://api.github.com").
 				Post("/repos/" + repositoryName + "/statuses").
-				SetMatcher(ExpectPayload(toHaveFailureState)).
+				SetMatcher(ExpectPayload(toBe(github.StatusFailure, "No tests in this PR :("))).
```
Comment (MatousJobanek) [inline pkg/plugin/test-keeper/plugin/event_handler_gh_test.go]: would make sense to use constants for these strings?

File: pkg/plugin/test-keeper/plugin/test_checker.go [hunk 2]
```diff
@@ -12,23 +12,48 @@ type TestChecker struct {
...
-func (checker *TestChecker) IsAnyNotExcludedFileTest(files []scm.ChangedFile) (bool, error) {
+// FileCategories holds information about the total files coming in the changeset, legit files (those which are excluded from test verification)
+// and Tests
+type FileCategories struct {
+	Total, Legit, Tests int
+	Files               *[]scm.ChangedFile
+}
+
+// OnlyLegitFiles indicates if changeset contains only files which are excluded from test verification
+func (f *FileCategories) OnlyLegitFiles() bool {
```
Comment (MatousJobanek) [inline pkg/plugin/test-keeper/plugin/test_checker.go]: aren't they more likely excluded or skipped files? Legit is (for me) something very vague and general.

File: pkg/plugin/test-keeper/plugin/test_checker.go [hunk 3]
```diff
@@ -12,23 +12,48 @@ type TestChecker struct {
...
+	return f.Total > 0 && f.Legit == f.Total
```
Comment (MatousJobanek) [inline pkg/plugin/test-keeper/plugin/test_checker.go]: do we really need to check the total count?
btw, if there is a PR without any file different, then it is automatically merged/closed

File: pkg/plugin/test-keeper/plugin/test_checker.go [hunk 4]
```diff
@@ -12,23 +12,48 @@ type TestChecker struct {
...
+
+// NewFileCategories creates new instance of FileCategories struct with files populated
+func NewFileCategories(files []scm.ChangedFile) FileCategories {
+	return FileCategories{Files: &files, Total: len(files)}
+}
 
-	remainingNoTestFiles := false
+// CategorizeFiles counts files in the changeset which are tests (included files) and should not be considered for
+// verification (excluded). When first test is found it stops, as this is enough to unblock PR
+func (checker *TestChecker) CategorizeFiles(files []scm.ChangedFile) (FileCategories, error) {
```
Comment (MatousJobanek) [inline pkg/plugin/test-keeper/plugin/test_checker.go]: Is this still `TestChecker` or something else? 
btw, when I read `CategorizeFiles` I would expect that I get some categories - slices of files separated into different categories, not only count of the file types. Would make sense to rename it?:
`TestChecker` -> `FileTypeCounter`
`CategorizeFiles` -> `Count`
`FileCategories` -> `FileTypes` or `CountedFileTypes`

I know that I'm not good at naming stuff, my proposal is maybe worse than yours, it just didn't make sense to me at the very first look.

File: pkg/plugin/test-keeper/plugin/event_handler_gh_test.go (hunk 1 above)
Comment (bartoszmajsak) [reply pkg/plugin/test-keeper/plugin/event_handler_gh_test.go]: Sure, good idea. On the way

File: pkg/plugin/test-keeper/plugin/test_checker.go (hunk 2 above)
Comment (bartoszmajsak) [reply pkg/plugin/test-keeper/plugin/test_checker.go]: Skipped is a good name! Thanks

File: pkg/plugin/test-keeper/plugin/test_checker.go (hunk 3 above)
Comment (bartoszmajsak) [reply pkg/plugin/test-keeper/plugin/test_checker.go]: There won't be 0 files in GitHub PR, that is true. But for the boundary case, if the changeset has 0 files than it have no tests and no skipped files - so answering to "onlySkippedFiles" should be false, right?

File: pkg/plugin/test-keeper/plugin/test_checker.go (hunk 4 above)
Comment (bartoszmajsak) [reply pkg/plugin/test-keeper/plugin/test_checker.go]: That's right, maybe `Categorize` is too much for just counting them up. Let me reword it.

File: pkg/plugin/test-keeper/plugin/test_checker.go (hunk 3 above)
Comment (MatousJobanek) [reply pkg/plugin/test-keeper/plugin/test_checker.go]: Agree. 
Last question  - do we need to also check the boundary case when there is no changed file in the PR - in the event_handler? Using common sense I would say no. As a pedantic guy probably yes :-D

File: pkg/plugin/test-keeper/plugin/event_handler_gh_test.go (hunk 1 above)
Comment (bartoszmajsak) [reply pkg/plugin/test-keeper/plugin/event_handler_gh_test.go]: def5e6c

Comment (bartoszmajsak) [reply pkg/plugin/test-keeper/plugin/event_handler_gh_test.go]: def5e6c

File: pkg/plugin/test-keeper/plugin/test_checker.go (hunk 2 above)
Comment (bartoszmajsak) [reply pkg/plugin/test-keeper/plugin/test_checker.go]: e5df495

File: pkg/plugin/test-keeper/plugin/test_checker.go (hunk 3 above)
Comment (bartoszmajsak) [reply pkg/plugin/test-keeper/plugin/test_checker.go]: That's unrealistic assuming it is coming from GitHub... So I would say YAGNI? ;)

File: pkg/plugin/test-keeper/plugin/test_checker.go (hunk 4 above)
Comment (bartoszmajsak) [reply pkg/plugin/test-keeper/plugin/test_checker.go]: e9377ce
